plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineEnclaveMetadata.kt

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * In native mode, enclave metadata is bundled into the enclave .so file by the Intel SGX SDK. The Intel SDK then uses
+ * this data when loading the enclave. This allows our code to be mostly agnostic of the enclave mode. With Gramine
+ * though, there is no concept of a "simulation" mode, so we have to implement some of this logic ourselves.
+ * This task produces a properties file which serves as our mechanism for passing build-time information for the host to
+ * use at runtime.
+ * It should be noted that this file is distinct from the enclave properties file in that it is available to the host
+ * *before* the enclave is loaded.
+ *
+ * TODO: Investigate using environment variables in the Manifest to pass this metadata instead.
+ */
+open class GenerateGramineEnclaveMetadata @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val METADATA_FILE_NAME = "enclave-metadata.properties"
+    }
+
+    @get:Input
+    val buildType: Property<BuildType> = objects.property(BuildType::class.java)
+    @get:Input
+    val maxThreads: Property<Int> = objects.property(Int::class.java)
+    @get:InputFile
+    val signingKey: RegularFileProperty = objects.fileProperty()
+    @Optional
+    @get:Input
+    val pythonEnclaveFile: Property<String> = objects.property(String::class.java)
+
+    @get:OutputFile
+    val outputGramineEnclaveMetadata: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val isSimulation = buildType.get() == BuildType.Simulation
+
+        /**
+         * We don't use the java Properties class here for two reasons:
+         * - It doesn't guarantee consistent order.
+         * - It prints the date and time at the top of the file.
+         */
+        val properties = TreeMap<String, String>().apply {
+            put("isSimulation", isSimulation.toString())
+            put("maxThreads", maxThreads.get().toString())
+
+            /**
+             * The signing key measurement should not be included in modes other than simulation.
+             * It should be noted that no actual signing takes place in simulation mode!
+             * TODO: CON-1163 you will probably need to remove this for mock attestation in debug/release mode!
+             */
+            if (isSimulation) {
+                val mrsigner = computeSigningKeyMeasurement(signingKey.asFile.get())
+                put("signingKeyMeasurement", Base64.getEncoder().encodeToString(mrsigner))
+            }
+
+            if (pythonEnclaveFile.isPresent) {
+                put("pythonEnclaveScript", pythonEnclaveFile.get())
+            }
+        }
+
+        outputGramineEnclaveMetadata.asFile.get().writer().use { writer ->
+            for (entry in properties.entries) {
+                writer.write("${entry.key}=${entry.value}\n")
+            }
+        }
+    }
+
+    /** Compute the mrsigner value from a provided .pem file containing a 3072 bit RSA key. */
```

Comment:
Is there documentation you can link to help the next person who might look at this code?